lib/RequestHandler.ts

```diff
@@ -0,0 +1,52 @@
+/// <reference path="../typings/tsd.d.ts" />
+import Promise = require('bluebird');
+import restify = require('restify');
+import apiSession = require('./api-session');
+import BAPI = require('./blpapi-wrapper');
+
+export = RequestHandler;
+
+class RequestHandler {
+    private blpSession: BAPI.Session;
+
+    constructor(blpsess: BAPI.Session) {
+        this.blpSession = blpsess;
```

Comment:
I moved the instantiation of `requesthandler` inside `createSession` so it would hold a substituted instance of `blpSession` in case `sessionTerminated`. Good catch, thanks!

I don't fully understand the use of `res.locals`. Are you suggesting tacking the `blpSession` onto the `req/res`? If so, I don't particular see any benefit of it since we are only use **1** session object. Is there any performance implication for using closure rather than direct property lookup? The only reason I am not using early binding is that `bind` will create an anonymous function, which would result in a `?` in restify's trace log.